### PR TITLE
chore: Update dev dependencies

### PR DESCRIPTION
**Pull request by edgarrmondragon**
## Summary

- Updated deptry config
- Updated docs lock file

## Test Plan

See if the tests and docs build pass.

## Checklist

- [x] My pull request has a descriptive title.
- [x] I have read the [CONTRIBUTING] guide.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] If appropriate, I have added necessary documentation.
- [ ] For user-facing changes, refactorings, performance improvements or documentation updates, I have added a changelog entry using the [Changie] tool.

For both the title of the PR and the changelog entry, prefer simple past tense or constructions with "now". For example:

  - Added `Client.invite_participants()`
  - `Client.user_activation_settings()` now accepts a `user_activation_settings` keyword argument

[CONTRIBUTING]: https://citric.readthedocs.io/en/latest/contributing/code-of-conduct.html
[Changie]: https://changie.dev/

----
📚 Documentation preview 📚: https://citric--1127.org.readthedocs.build/en/1127/

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -22,13 +22,12 @@ Release **v{sub-ref}`version`**. ([What's new?](./changelog.md))
...
-- {ls_tag}`6.4.2+240115`
-- {ls_tag}`5.6.53+240131`
-- {ls_tag}`5.6.52+240123`
-- {ls_tag}`5.6.51+240116`
+- {ls_tag}`6.5.1+240320`
+- {ls_tag}`6.5.0+240319`
+- {ls_tag}`6.4.11+240304`
+- {ls_tag}`5.6.57+240312`
+- {ls_tag}`5.6.56+240227`
+- {ls_tag}`5.6.55+240220`
```
Comment (edgarrmondragon) [inline docs/index.md:30]: Updates the list of versions we know are passing in integration tests in this PR

https://citric.readthedocs.io/en/latest/#integration-tests

File: .github/workflows/constraints.txt [hunk 2]
```diff
@@ -2,4 +2,4 @@ griffe==0.42.1
 nox==2024.3.2
 pip==24.0
 pip-tools==7.4.1
-uv==0.1.26
+uv==0.1.31
```
Comment (edgarrmondragon) [inline .github/workflows/constraints.txt:5]: uv is a Python package installer

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -14,7 +14,7 @@
 - [ ] I have read the [CONTRIBUTING] guide.
 - [ ] I have added tests that prove my fix is effective or that my feature works.
 - [ ] If appropriate, I have added necessary documentation.
-- [ ] For user-facing changes, refactorings, performance improvements or documentation updates, I have added a changelog entry using the [Changie] tool.
+- [ ] For user-facing changes, refactorings, performance improvements or documentation updates, I have added a changelog entry using [Changie].
```
Comment (edgarrmondragon) [inline .github/pull_request_template.md:17]: This just feels like it's better worded

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -8,7 +8,7 @@ ci:
 
 repos:
 - repo: https://github.com/pre-commit/pre-commit-hooks
-  rev: v4.5.0
+  rev: v4.6.0
```
Comment (edgarrmondragon) [inline .pre-commit-config.yaml:11]: These ase some useful quality assurance checks that make sure files have the right structure, dependencies are up to date, etc.

**Comment by edgarrmondragon** [comment]
Ping @GabxTaro :)

**Review by GabxTaro — APPROVED** [review]

**Pull request merged**